src/stan/services/init/initialize_state.hpp

```diff
@@ -113,52 +114,44 @@ namespace stan {
        *                            right size and set to 0.
        * @param[in,out] model       the model. Side effects on model? I'm not
        *                            quite sure
-       * @param[in,out] output      output stream for messages
+       * @param[in,out] writer      writer callback for messages
        */
       template <class Model>
       bool initialize_state_values(Eigen::VectorXd& cont_params,
                                    Model& model,
-                                   std::ostream* output) {
+                                   base_writer& writer) {
         try {
           validate_unconstrained_initialization(cont_params, model);
         } catch (const std::exception& e) {
-          if (output)
-            *output << e.what() << std::endl;
+          writer(e.what());
           return false;
         }
         double init_log_prob;
         Eigen::VectorXd init_grad = Eigen::VectorXd::Zero(model.num_params_r());
         try {
           stan::model::gradient(model, cont_params, init_log_prob,
-                                init_grad, output);
+                                init_grad);
```

Comment:
Gradient passes errors through an exception so no output should be needed.  May want to double check.